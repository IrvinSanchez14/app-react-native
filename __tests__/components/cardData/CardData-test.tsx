/**
 * @format
 */

import React from 'react';
import CardData from '_components/cardData';
import { View } from 'react-native';
import { Card, Text, Icon } from 'native-base';
import { Collapse, CollapseHeader, CollapseBody } from 'accordion-collapse-react-native';
import { shallow, ShallowWrapper } from 'enzyme';

jest.mock('react-navigation', () => {
	return {
		createAppContainer: jest.fn().mockReturnValue(function NavigationContainer() {
			return null;
		}),
		createDrawerNavigator: jest.fn(),
		createMaterialTopTabNavigator: jest.fn(),
		createStackNavigator: jest.fn(),
		StackActions: {
			push: jest.fn().mockImplementation(x => ({ ...x, type: 'Navigation/PUSH' })),
			replace: jest.fn().mockImplementation(x => ({ ...x, type: 'Navigation/REPLACE' })),
		},
		NavigationActions: {
			navigate: jest.fn().mockImplementation(x => x),
		},
	};
});

describe('CardData Component', () => {
	describe('Rendering', () => {
		let wrapper: ShallowWrapper;
		beforeEach(() => {
			wrapper = shallow(<CardData />);
		});
		test('Check if render component correctly', () => {
			expect(wrapper).toMatchSnapshot();
		});
		test('Should render a <Card />', () => {
			expect(wrapper.find(Card)).toHaveLength(1);
		});
		test('Should render a <Collapse />', () => {
			expect(wrapper.find(Card).find(Collapse)).toHaveLength(1);
		});
		test('Should render a 3 <View />', () => {
			expect(
				wrapper
					.find(Card)
					.find(Collapse)
					.find(View)
			).toHaveLength(3);
		});
		test('Should render a 6 <Icon />', () => {
			expect(
				wrapper
					.find(Card)
					.find(Collapse)
					.find(View)
					.find(Icon)
			).toHaveLength(6);
		});
		test('Should render a 6 <Text />', () => {
			expect(
				wrapper
					.find(Card)
					.find(Collapse)
					.find(View)
					.find(Text)
			).toHaveLength(6);
		});
		test('collapse return false when state change onToggle', () => {
			// Arrange
			//const mockOnPress = jest.fn(); // 1. mock function
			//wrapper.find('#collapseID').simulate('onPress', mockOnPress(true));
			//console.log(wrapper.find('#collapseID').prop('isCollapsed'));
			expect(wrapper.find('#collapseID').prop('isCollapsed')).toBe(false);
		});
	});
});
