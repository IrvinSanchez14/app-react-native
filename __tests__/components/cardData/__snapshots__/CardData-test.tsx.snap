// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardData Component Rendering Check if render component correctly 1`] = `
<Styled(Card)
  style={
    Object {
      "marginTop": 0,
    }
  }
>
  <Collapse
    disabled={false}
    id="collapseID"
    isCollapsed={false}
    onToggle={[Function]}
  >
    <CollapseHeader
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#EFEFEF",
          "height": 82,
        }
      }
    >
      <View
        style={
          Object {
            "height": 100,
            "marginLeft": 14,
            "marginTop": 10,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "color": "#548B6F",
              "fontWeight": "bold",
              "marginTop": 10,
            }
          }
        >
          94378331
        </Styled(Text)>
        <Styled(Text)>
          JOE LOCKWOOD
        </Styled(Text)>
        <Styled(Icon)
          name="file-text"
          style={
            Object {
              "fontSize": 20,
              "marginRight": 140,
              "marginTop": -35,
              "textAlign": "right",
            }
          }
          type="FontAwesome"
        />
        <Styled(Icon)
          name="truck"
          style={
            Object {
              "fontSize": 20,
              "marginRight": 93,
              "marginTop": -20,
              "textAlign": "right",
            }
          }
          type="FontAwesome5"
        />
        <Styled(Icon)
          name="shopping-cart"
          style={
            Object {
              "color": "#548B6F",
              "fontSize": 20,
              "marginRight": 55,
              "marginTop": -23,
              "textAlign": "right",
            }
          }
          type="FontAwesome5"
        />
        <Styled(Icon)
          name="chevron-down"
          style={
            Object {
              "color": "#B7CBC3",
              "fontSize": 20,
              "marginRight": 21,
              "marginTop": -23,
              "textAlign": "right",
            }
          }
          type="FontAwesome5"
        />
      </View>
    </CollapseHeader>
    <CollapseBody>
      <View
        style={
          Object {
            "marginTop": 15,
          }
        }
      >
        <Styled(Icon)
          name="clockcircle"
          style={
            Object {
              "color": "#548B6F",
              "fontSize": 20,
              "marginLeft": 15,
              "marginTop": 13,
            }
          }
          type="AntDesign"
        />
        <Styled(Text)
          style={
            Object {
              "color": "#548B6F",
              "fontWeight": "bold",
              "marginLeft": 45,
              "marginTop": -22,
            }
          }
        >
          Order placed
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "marginLeft": 45,
            }
          }
        >
          4 hours ago
        </Styled(Text)>
      </View>
      <View
        style={
          Object {
            "height": 150,
          }
        }
      >
        <Styled(Icon)
          name="file-text"
          style={
            Object {
              "color": "#548B6F",
              "fontSize": 20,
              "marginLeft": 15,
              "marginTop": 13,
            }
          }
          type="FontAwesome"
        />
        <Styled(Text)
          style={
            Object {
              "color": "#548B6F",
              "fontWeight": "bold",
              "marginLeft": 45,
              "marginTop": -22,
            }
          }
        >
          Special instructions
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "marginLeft": 45,
            }
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sed varius nisi, id tincidunt ligula. Suspendisse vestibulum erat molestie elit suscipit, volutpat pellentesque dui euismod.
        </Styled(Text)>
      </View>
    </CollapseBody>
  </Collapse>
</Styled(Card)>
`;
